design-proposals/vpro-device.md

```diff
@@ -0,0 +1,196 @@
+# Design Proposal: vPRO/AMT/ISM devices activation
+
+Author(s): Edge Infrastructure Manager Team
+
+Last updated: 05/12/2025
+
+## Abstract
+
+vPRO/Active Management Technology (AMT)/Intel Standard Manageability (ISM) needs to be explicitily activated and
+configured in the devices before to be available in the Edge Orchestrator.
+
+[Open Device Management Toolkit](https://device-management-toolkit.github.io/docs/2.27/GetStarted/overview/) (Open DMT)
+provides an open source stack through which is possible to manage vPRO enabled devices.
+
+![EMF Stack](./images/specs/stack.svg)
+
+This document describes the design proposal for adding the Remote Provisioning Client (RPC) component of DMT stack into
+the EN sw and propose a design on how to manage the device activation during the EN journey.
+
+## Proposal
+
+Remote Provisioning Client (RPC) is launched during the provisioning flow (uOS stage).
+
+vPRO (AMT & ISM) capable Core devices need to have the `rpc-go` utility embedded in the uOS image (the so called
+Tink-EMT image - replacement of HookOS), as this is required to activate AMT. This also requires the `heci` driver to
+be included in the kernel as the `rpc-go` utility communicates with CSME over PCIe/HECI in to activate the device.
+
+`Local Manageability Service` (LMS) must to be included as it is still required to enable the communication between RPC
+and AMT device. Additionally, it offers the support for in-band commands too.
+
+Adding the `rpc-go` to the Tink-EMT image allows for the AMT/ISM device to be activated independently of the target
+operating system being deployed on the edge node. Not every device will support AMT & ISM. Introspection of the device
+capabilites will be added to the `Device Discovery` to determine:
+
+1. AMT Supported Device
+2. ISM Supported Device
+3. No OOB Supported
+
+This will also allow for a custom workflows in Tinkerbell to be used based on whether the device support AMT or ISM.
+
+If an attempt is made to activate an unsupported or faulty device, the `rpc-go` client will yield an error. The
+tinkerbell workflow should appropriately be handling the common error scenarios, please refer to the official
+[DMT/rpc-go documentation](https://device-management-toolkit.github.io/docs/2.27/Reference/RPC/libraryRPC/#rpc-error-code-charts)
+
+Let us now analyze the device activation, the user must do the following steps before starting AMT activation
+
+- AMT is enabled in BIOS
```

Comment:
what is the default for this ? enabled or disabled ? (say when the device come fresh from factory ?)

Reply:
like SB and FDE we dont know - it is really dependent on the vendor. This will be part of our docs